### ISSUE TITLE
Android build failing with 2.1.3 or 2.2.1 version of the plugin

### ISSUE THREAD

**Issue by pitAlex** [phonegap/phonegap-plugin-push#2256]
Hi,
So today I did some updates on the app as I learned about cordova-android 7.1.0 and a new version of the push plugin. As I tried to update the app but build kept failing and I wanted to revert to android 7.0 with 2.1.3. But even so its still failing with the following:

Exception in thread "main" java.lang.IllegalArgumentException
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:108)
        at com.google.devtools.build.android.desugar.DefaultMethodClassFixer$DefaultMethodFinder.visit(DefaultMethodClassFixer.java:471)
        at org.objectweb.asm.ClassReader.accept(Unknown Source)
        at org.objectweb.asm.ClassReader.accept(Unknown Source)
        at com.google.devtools.build.android.desugar.DefaultMethodClassFixer.defaultMethodsDefined(DefaultMethodClassFixer.java:319)
        at com.google.devtools.build.android.desugar.DefaultMethodClassFixer.visitEnd(DefaultMethodClassFixer.java:88)
        at org.objectweb.asm.ClassVisitor.visitEnd(Unknown Source)
        at com.google.devtools.build.android.desugar.InterfaceDesugaring.visitEnd(InterfaceDesugaring.java:85)
        at org.objectweb.asm.ClassVisitor.visitEnd(Unknown Source)
        at com.google.devtools.build.android.desugar.LambdaDesugaring.visitEnd(LambdaDesugaring.java:150)
        at org.objectweb.asm.ClassReader.accept(Unknown Source)
        at org.objectweb.asm.ClassReader.accept(Unknown Source)
        at com.google.devtools.build.android.desugar.Desugar.desugarClassesInInput(Desugar.java:401)
        at com.google.devtools.build.android.desugar.Desugar.desugarOneInput(Desugar.java:326)
        at com.google.devtools.build.android.desugar.Desugar.desugar(Desugar.java:280)
        at com.google.devtools.build.android.desugar.Desugar.main(Desugar.java:584)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformClassesWithDesugarForDebug'.

I've removed almost all plugins and the above error only happens when I have the push plugin added.
Whats going on here and how can I resolve this?

I would like to update to say that reverting to 2.1.2 with android 7.0.0 made the build complete again.

**Comment by peterpeterparker** [comment]
I faced the same problem a couple of days ago, see https://github.com/phonegap/phonegap-plugin-push/issues/2247

My solution was an incompatibility between the FCM version of push plugin and analytics plugin. I changed the analytics one and everything went back to normal. As described in the issue I linked, it's possible to specify another FCM version thru a variable when you add the push plugin.
